### PR TITLE
Support for loading in-memory files

### PR DESCRIPTION
**Pull request by OMantere**
This PR adds a small wrapper for `np.fromfile` to allow loading of BytesIO wrapped FCS files by using a new call `FCSParser.fromfile()`. Also abstracting the dataframe constructing code to a method to enable getting that from the `FCSParser` object.

### CODE REVIEW

**Comment by eyurtsev** [comment]
Thanks for the PR -- I'll try and review within a few days -- Could you fix the accidental commit in the tests file? It's causing the tests to fail

**Comment by OMantere** [comment]
@eyurtsev Sorry about that. Should be fixed now. Thanks for taking the time to review.

File: fcsparser/api.py [hunk 1]
```diff
@@ -430,6 +444,12 @@ def reformat_meta(self):
         meta['_channels_'] = df
         meta['_channel_names_'] = self.get_channel_names()
 
+    # Constructs Pandas dataframe
+    def dataframe(self):
```
Comment (eyurtsev) [inline fcsparser/api.py]: okay if I change method name to_dataframe() else change the method to a property?

(I like method names to be verbs with nouns being reserved for attributes, classes and properties)

File: fcsparser/api.py [hunk 2]
```diff
@@ -95,20 +95,34 @@ def __init__(self, path, read_data=True, channel_naming='$PnS'):
         self._data_end = -1
         self.channel_numbers = []
         self._analysis = ''
-
-        self._file_size = os.path.getsize(path)
+        self._file_size = 0
```
Comment (eyurtsev) [inline fcsparser/api.py]: Thanks this is much cleaner

File: fcsparser/api.py [hunk 3]
```diff
@@ -95,20 +95,34 @@ def __init__(self, path, read_data=True, channel_naming='$PnS'):
...
+        file_handle.seek(0)
+        self.read_header(file_handle)
+        self.read_text(file_handle)
+        if read_data:
+            self.read_data(file_handle)
+
+    @staticmethod
+    def from_data(data):
+        '''Loads an FCS file from a bytes-like object'''
+        file_handle = BytesIO(data)
```
Comment (eyurtsev) [inline fcsparser/api.py]: Is a context manager needed for BytesIO?

```
with BytesIO(data) as file_handle:
  obj.load_file(file_handle)
return obj
```

or 

```
with contextlib.closing(BytesIO(data)) as file_handle:
```

Comment (OMantere) [inline fcsparser/api.py]: @eyurtsev Seems to work either way, I'm fine with both ways.

File: fcsparser/api.py (hunk 1 above)
Comment (OMantere) [inline fcsparser/api.py]: @eyurtsev Sure, I like the idea of a property the best, but again, I'm fine either way.

**Comment by OMantere** [comment]
@eyurtsev Can we push the latest version to PyPi? Thanks.

**Comment by eyurtsev** [comment]
working on it now -- sorry for delay -- as usual totally blanked out

**Comment by eyurtsev** [comment]
Upload to pypi let me know if you bump into any issues.

FYI -- I changed the `dataframe` method to a property, so you may need to update some of your code

**Pull request merged**